gno.land/pkg/gnoweb/url_test.go

```diff
@@ -0,0 +1,130 @@
+package gnoweb
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseGnoURL(t *testing.T) {
+	testCases := []struct {
+		Name     string
+		Input    string
+		Expected *GnoURL
+		Err      *error
+	}{
+		{
+			Name:     "malformed url",
+			Input:    "https://gno.land/r/dem)o:$?",
+			Expected: nil,
+			Err:      &ErrURLMalformedPath,
+		},
+
+		{
+			Name:  "webquery + query",
+			Input: "https://gno.land/r/demo/foo$help&func=Bar&name=Baz",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "",
+				WebQuery: url.Values{
+					"help": []string{""},
+					"func": []string{"Bar"},
+					"name": []string{"Baz"},
+				},
+				Query: url.Values{},
+				Host:  "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "path args + webquery",
+			Input: "https://gno.land/r/demo/foo:example$tz=Europe/Paris",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "example",
+				WebQuery: url.Values{
+					"tz": []string{"Europe/Paris"},
+				},
+				Query: url.Values{},
+				Host:  "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "path args + webquery + query",
+			Input: "https://gno.land/r/demo/foo:example$tz=Europe/Paris?hello=42",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "example",
+				WebQuery: url.Values{
+					"tz": []string{"Europe/Paris"},
+				},
+				Query: url.Values{
+					"hello": []string{"42"},
+				},
+				Host: "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "webquery inside query",
+			Input: "https://gno.land/r/demo/foo:example?value=42$tz=Europe/Paris",
+			Expected: &GnoURL{
+				Path:     "/r/demo/foo",
+				Kind:     KindRealm,
+				Args:     "example",
+				WebQuery: url.Values{},
+				Query: url.Values{
+					"value": []string{"42$tz=Europe/Paris"},
+				},
+				Host: "gno.land",
+			},
+			Err: nil,
+		},
```

Comment:
So, the gnoweb query and the realm query need to be in a particular order, ie 
`pkgpath:renderpath$tz=Europe/Paris?realmquerykey=value&newvalue=1`?

gnoweb query needs to be before the realm query?

Reply:
Yes, the order matters. More specifically, we don't want to have webquery within the query because it would make the URL non-compliant with standard URL format, and most common REST tools wouldn't be able to correctly manipulate this kind of URL, i.e., Postman. It would also make parsing and error management more complex to implement.
c.f https://github.com/gnolang/gno/issues/3162